### DISCUSSION TITLE
RSTP to HLS (audio + video)

### DISCUSSION THREAD

**Discussion by Dmise** [bluenviron/mediamtx#1717] [Questions] [answered]
Hello. The task is common. I have rtsp stream from camera, and I want to restream it to the browser (one page in developing web application). 

I have launched mediamtx server (the host is windows 11, my work laptop). Configure .yml file, point rtsp source (has audio+video), try go het HLS from server. But mediamtx create HLS with video track only. 

question:  how to get HLS with audio and video. Without using gstreamer or ffmpeg commands ( so to say from the box), or I have to use ffmpeg or gstreamer commands on demand.
 

**iframe tag where i want to see camera (video+audio):**
```bash
<iframe src="http://127.0.0.1:8888/stream1" width="1280" height="100%" scrolling="no"></iframe>
```

I've tried all audio formats available on camera setting. In all cases mediamtx just cut audio, and remain only video track for HLS stream
**audio codecs on ip camera**
![audio codecs](https://user-images.githubusercontent.com/46092536/232481833-8d814ece-7dd7-44cf-8b32-54cec50439ea.png)
**mediamtx output**
```bash
PS A:\Distr\mediamtx> ./mediamtx
2023/04/17 14:24:15 INF MediaMTX / rtsp-simple-server v0.22.1
2023/04/17 14:24:15 INF [path stream1] [rtsp source] started
2023/04/17 14:24:15 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/04/17 14:24:15 INF [RTMP] listener opened on :1935
2023/04/17 14:24:15 INF [HLS] listener opened on :8888
2023/04/17 14:24:15 INF [WebRTC] listener opened on :8889 (HTTP)
2023/04/17 14:24:15 INF [API] listener opened on 127.0.0.1:9997
2023/04/17 14:24:15 INF [path stream1] [rtsp source] ready: 3 tracks (H264, MPEG2-audio, Generic)
2023/04/17 14:24:15 INF [HLS] [muxer stream1] created automatically
2023/04/17 14:24:15 INF [HLS] [muxer stream1] is converting into HLS, 1 track (H264)
2023/04/17 14:33:04 INF [path stream1] [rtsp source] ERR: read tcp 192.168.1.102:55848->192.168.1.64:554: wsarecv: An existing connection was forcibly closed by the remote host.
2023/04/17 14:33:04 INF [HLS] [muxer stream1] destroyed (terminated)
2023/04/17 14:33:09 INF [path stream1] [rtsp source] ready: 3 tracks (H264, G711, Generic)
2023/04/17 14:33:09 INF [HLS] [muxer stream1] created automatically
2023/04/17 14:33:09 INF [HLS] [muxer stream1] is converting into HLS, 1 track (H264)
```

[mediamtx.yml.txt](https://github.com/aler9/mediamtx/files/11249562/mediamtx.yml.txt)

**Comment by aler9** [comment, marked as answer]
Hello, as clearly written in the README, HLS only supports audio encoded in AAC or Opus. Both codecs are not supported by your camera. You have to re-encode audio in AAC by using FFmpeg.
I've created a feature request in order to support reading MP3 audio with AAC: #1719

**Reply by Dmise** [reply]
Got you. Will wait update of mediamtx server. Now, all that remains its use ffmpeg.

[a comment by an automated account is not shown]
